src/astro/sql/operators/sql_decorator.py

```diff
@@ -218,29 +218,22 @@ def handle_output_table_schema(self, output_table_name, schema=None):
         :return:
         """
         schema = schema or get_schema()
-        if self.conn_type == "postgres" and self.schema:
+        if self.conn_type in ["postgres", "bigquery"] and self.schema:
             output_table_name = schema + "." + output_table_name
         elif self.conn_type == "snowflake" and self.schema and "." not in self.sql:
             output_table_name = self.database + "." + schema + "." + output_table_name
         return output_table_name
 
     def _set_schema_if_needed(self):
         schema_statement = ""
-        if self.conn_type == "postgres":
+        if self.conn_type in ["postgres", "snowflake", "bigquery"]:
             schema_statement = create_schema_query(
                 conn_type=self.conn_type,
                 hook=self.hook,
                 schema_id=self.schema,
                 user=self.user,
             )
-        elif self.conn_type == "snowflake":
```

Comment:
Good refactor! :tada: 
I wonder if we could be abstracting some of this logic as part of the `Table` class, and have some method e.g. `has_schema`, so we avoid having to hard-code this type of list in multiple places.